src/App.js

```diff
@@ -1,18 +1,73 @@
 // TODO: translate everything to SLOVAK language
+// TODO: add dark mode to all pages
+// TODO: make dark mode switching more smooth
+// TODO: generate MIT license here: https://license-generator.intm.org/
 
-import { BrowserRouter as Router, Route } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import { useDefaultValue } from './hooks/useDefaultValue';
+import { load as loadCookie, save as saveCookie } from 'react-cookies';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
 import { Home } from './components/Home';
 import { Login } from './components/Login';
+import { _404 } from './components/404';
+import { About } from './components/About';
+import './styles/Global.css';
+
+// got a better idea? write it down here
+let useTheme = className => className;
+
+const App = () => {
+    const [token, setToken] = useState(loadCookie("token"));  // this will return UNDEFINED if its not in cookies
+    const [darkMode, setDarkMode] = useState(useDefaultValue(
+        loadCookie("dark-mode") === 'true',
+        window.matchMedia("(prefers-color-scheme: dark)").matches
+    ));
+    const [snowFlakes, setSnowFlakes] = useState(useDefaultValue(
+        loadCookie('snowflakes') === 'true',
+        true
+    ));
+
+    useTheme = (className) => {
+        return `${className} ${darkMode ? `${className}-dark` : ""}`
+    }
+
+    useEffect(() => {
+        saveCookie("dark-mode", darkMode);
+    }, [darkMode]);
 
-function App() {
     return (
         <Router>
             <div>
-                <Route path="/" exact={ true } component={ Home } />
-                <Route path="/login" exact={ true } component={ Login } />
+                <Switch>
+                    <Route path="/" exact={ true } render={ routeProps => (
+                        <Home
+                            token={ token }
+                            darkMode={ darkMode }
+                            setDarkMode={ setDarkMode }
+                            snowFlakes={ snowFlakes }
+                            setSnowFlakes={ setSnowFlakes }
+                        />
+                    )} />
+
+                    <Route path="/login" exact={ true } render={ routeProps => (
+                        <Login
+                            setToken={ setToken }
+                            darkMode={ darkMode }
+                        />
+                    )} />
+
+                    <Route path="/about" exact={ true } render={ routeProps => (
+                        <About />
+                    )} />
+
+                    <Route render={ routeProps => (
+                        // eslint-disable-next-line
+                        <_404 />
```

Comment:
You can call this component something like `NotFoundErrorPage` to avoid this warning.

Reply:
Called it `NotFoundPage`